packages/calcite-components/src/components/panel/panel.tsx

```diff
@@ -501,7 +491,7 @@ export class Panel extends LitElement implements InteractiveComponent {
         ariaLabel={close}
         icon={ICONS.close}
         id={IDS.close}
-        onClick={this.handleUserClose}
+        onClick={this.emitCloseEvent}
```

Comment:
Can we use a dedicated `click` event handler here instead? This keeps event handling separate from emitting the event and helps avoid mixing both concerns if additional logic gets added later.